src/test/java/org/isf/medicalstock/test/Tests.java

```diff
@@ -73,108 +87,213 @@
-	@Autowired
-	MedicalsIoOperationRepository medicalsIoOperationRepository;
-	@Autowired
-	MedicalTypeIoOperationRepository medicalTypeIoOperationRepository;
-	@Autowired
-	WardIoOperationRepository wardIoOperationRepository;
-	@Autowired
-	MovementIoOperationRepository movementIoOperationRepository;
-	@Autowired
-	MedicalStockMovementTypeIoOperationRepository medicalStockMovementTypeIoOperationRepository;
-	@Autowired
-	SupplierIoOperationRepository supplierIoOperationRepository;
-	@Autowired
-	ApplicationEventPublisher applicationEventPublisher;
+	private EntityManager entityManager;
 
-	@BeforeClass
-	public static void setUpClass() {
-		jpa = new DbJpaUtil();
-		testLot = new TestLot();
-		testMovement = new TestMovement();
-		testMedical = new TestMedical();
-		testMedicalType = new TestMedicalType();
-		testMovementType = new TestMovementType();
-		testWard = new TestWard();
-		testSupplier = new TestSupplier();
-	}
+    @Autowired
+    MedicalStockIoOperations medicalStockIoOperation;
 
-	@Before
-	public void setUp() throws OHException {
-		cleanH2InMemoryDb();
-		jpa.open();
-		testLot.setup(false);
-	}
+    public Tests(boolean in, boolean out, boolean toward) {
+	    GeneralData.AUTOMATICLOT_IN = in;
+	    GeneralData.AUTOMATICLOT_OUT = out;
+	    GeneralData.AUTOMATICLOTWARD_TOWARD = toward;
+    }
 
-	@After
-	public void tearDown() throws Exception
-	{
-		jpa.flush();
-		jpa.close();
+	@BeforeClass
+    public static void setUpClass()  
+    {
+    	jpa = new DbJpaUtil();
+    	testLot = new TestLot();
+    	testMovement = new TestMovement();
+    	testMedical = new TestMedical();
+    	testMedicalType = new TestMedicalType();
+    	testMovementType = new TestMovementType();
+    	testWard = new TestWard();
+    	testSupplier = new TestSupplier();
+    }
+
+    @Before
+    public void setUp() throws OHException
+    {
+        jpa.open();
+        
+        testLot.setup(false);
+    }
+        
+    @After
+    public void tearDown() throws Exception 
+    {
+    	cleanH2InMemoryDB();
+        jpa.close();
+    }
+    
+    @AfterClass
+    public static void tearDownClass() throws OHException 
+    {
+    	testLot = null;
+    	testMovement = null;
+    	testMedical = null;
+    	testMedicalType = null;
+    	testMovementType = null;
+    	testWard = null;
+    	testSupplier = null;
+    }
+
+	@Parameterized.Parameters(name ="Test with AUTOMATICLOT_IN={0}, AUTOMATICLOT_OUT={1}, AUTOMATICLOTWARD_TOWARD={2}")
+	public static Collection<Object[]> automaticlot() {
+		return Arrays.asList(new Object[][] {
+				{ false, false, false },
+				{ false, false, true },
+				{ false, true, false },
+				{ false, true, true },
+				{ true, false, false },
+				{ true, false, true },
+				{ true, true, false },
+				{ true, true, true }
+		});
 	}
 
 	@Test
-	public void testLotGets() throws Exception {
-		String code = _setupTestLot(false);
-		_checkLotIntoDb(code);
+	public void testLotGets()  throws Exception	{
```

Comment:
These changes are undoing all the recent changes to simplify the tests; please see
https://github.com/informatici/openhospital-core/blob/develop/src/test/java/org/isf/medicalstock/test/Tests.java

Note that all the try-catch-fail code has been removed in all the individual tests.

Reply:
Thank you, I had not noticed what you meant, I have put back the changes, but now there are 4 tests failing. Maybe you may help me to figure out why


Results :

```
Tests in error:
  testIoNewAutomaticDischargingMovementDifferentLots[Test with AUTOMATICLOT_IN=true, AUTOMATICLOT_OUT=false, AUTOMATICLOTWARD_TOWARD=false](org.isf.medicalstock.test.Tests): org.springframework.orm.jpa.JpaSystemException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 6306411973620952638; nested exception is org.hibernate.HibernateException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 6306411973620952638
  testIoNewAutomaticDischargingMovementDifferentLots[Test with AUTOMATICLOT_IN=true, AUTOMATICLOT_OUT=false, AUTOMATICLOTWARD_TOWARD=true](org.isf.medicalstock.test.Tests): org.springframework.orm.jpa.JpaSystemException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 3468854848434020106; nested exception is org.hibernate.HibernateException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 3468854848434020106
  testIoNewAutomaticDischargingMovementDifferentLots[Test with AUTOMATICLOT_IN=true, AUTOMATICLOT_OUT=true, AUTOMATICLOTWARD_TOWARD=false](org.isf.medicalstock.test.Tests): org.springframework.orm.jpa.JpaSystemException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 3674108227381538889; nested exception is org.hibernate.HibernateException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 3674108227381538889
  testIoNewAutomaticDischargingMovementDifferentLots[Test with AUTOMATICLOT_IN=true, AUTOMATICLOT_OUT=true, AUTOMATICLOTWARD_TOWARD=true](org.isf.medicalstock.test.Tests): org.springframework.orm.jpa.JpaSystemException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 4334341864990172666; nested exception is org.hibernate.HibernateException: identifier of an instance of org.isf.medicalstock.model.Lot was altered from 123456 to 4334341864990172666
```

Reply:
This test file has been problematic for a long time.  Note that there are still some `jpa` usage in the file because I couldn't get it work (see this [PR comment](https://github.com/informatici/openhospital-core/pull/243#issue-521843750).  I've tried a few things in this current PR without success.  Maybe @tomek39856 can lend a hand; he seems adapt at figuring out issues like this.

Reply:
> This test file has been problematic for a long time. Note that there are still some `jpa` usage in the file because I couldn't get it work (see this [PR comment](https://github.com/informatici/openhospital-core/pull/243#issue-521843750). I've tried a few things in this current PR without success. Maybe @tomek39856 can lend a hand; he seems adapt at figuring out issues like this.

Thanks @dbmalkovsky , it is finally fixed

CC @mwithi 

```
Results :

Tests run: 564, Failures: 0, Errors: 0, Skipped: 1

[INFO]
[INFO] --- maven-jar-plugin:3.2.0:jar (default-jar) @ OH-core ---
[INFO] Building jar: \openhospital\core\target\OpenHospital20\bin\OH-core.jar
[INFO]
[INFO] --- maven-jar-plugin:3.2.0:test-jar (default) @ OH-core ---
[INFO] Building jar: \openhospital\core\target\OpenHospital20\bin\OH-core-tests.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  44.536 s
[INFO] Finished at: 2020-11-27T20:30:41-05:00
[INFO] ------------------------------------------------------------------------
```